cmake/cpp_bindgen.cmake.in

```diff
@@ -65,18 +65,23 @@ function(bindgen_enable_fortran_library target_name)
     set(__CPP_BINDGEN_CMAKE_DIR @__CPP_BINDGEN_CMAKE_DIR@)
 
     if(CMAKE_Fortran_COMPILER_LOADED)
-        if(NOT TARGET fortran_bindings_handle)
-            add_library(fortran_bindings_handle ${__CPP_BINDGEN_SOURCE_DIR}/cpp_bindgen/array_descriptor.f90 ${__CPP_BINDGEN_SOURCE_DIR}/cpp_bindgen/handle.f90)
-            target_link_libraries(fortran_bindings_handle PUBLIC cpp_bindgen_handle)
-            target_include_directories(fortran_bindings_handle PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
-            include(${__CPP_BINDGEN_CMAKE_DIR}/fortran_helpers.cmake)
-            bindgen_enable_fortran_preprocessing_on_target(fortran_bindings_handle)
+        if(NOT TARGET cpp_bindgen_fortran_handle)
+            add_library(cpp_bindgen_fortran_handle ${__CPP_BINDGEN_SOURCE_DIR}/cpp_bindgen/array_descriptor.f90 ${__CPP_BINDGEN_SOURCE_DIR}/cpp_bindgen/handle.f90)
+            set(CPP_BINDGEN_Fortran_MODULES_DIRECTORY ${CMAKE_BINARY_DIR}/cpp_bindgen_modules
+                CACHE INTERNAL "Directory for Fortran modules of cpp_bindgen")
+            set_target_properties(cpp_bindgen_fortran_handle PROPERTIES Fortran_MODULE_DIRECTORY ${CPP_BINDGEN_Fortran_MODULES_DIRECTORY})
+            target_link_libraries(cpp_bindgen_fortran_handle PUBLIC cpp_bindgen_handle)
+            target_include_directories(cpp_bindgen_fortran_handle PUBLIC $<BUILD_INTERFACE:${CPP_BINDGEN_Fortran_MODULES_DIRECTORY}> $<INSTALL_INTERFACE:include/>)
+            include(${__CPP_BINDGEN_CMAKE_DIR}/cpp_bindgen_fortran_helpers.cmake)
+            bindgen_enable_fortran_preprocessing_on_target(cpp_bindgen_fortran_handle)
         endif()
         if(NOT TARGET ${target_name}_fortran)
             set_source_files_properties(GT_${${target_name}_fortran_bindings_path} PROPERTIES GENERATED TRUE)
-            add_library(${target_name}_fortran EXCLUDE_FROM_ALL ${GT_${target_name}_fortran_bindings_path})
+            add_library(${target_name}_fortran EXCLUDE_FROM_ALL ${CPP_BINDGEN_${target_name}_fortran_bindings_path})
             target_link_libraries(${target_name}_fortran PUBLIC ${target_name})
-            target_link_libraries(${target_name}_fortran PUBLIC fortran_bindings_handle)
+            target_link_libraries(${target_name}_fortran PUBLIC cpp_bindgen_fortran_handle)
+            # location of .mod file (we cannot know the INSTALL_INTERFACE directory here)
+            target_include_directories(${target_name}_fortran PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
```

Comment:
Yes, that's what CMake promises (if `CMAKE_Fortran_MODULE_DIR` and `Fortran_MODULE_DIR` of the target properties are not set).